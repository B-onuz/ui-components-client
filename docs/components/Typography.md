# Typography (Component)

Componente para textos em geral.

## Como utilizar

```js
import Typography from 'ui-components/lib/components/Typography'
```

```jsx
<Typography>Laboris ut veniam sit veniam eiusmod dolor quis consequat labore cillum esse cillum occaecat incididunt.</Typography>
```

## Variações

### Variant

```jsx
<Typography variant={'variant'} />
```

Escala no tema: `theme.variantScale`

## Styled-system props

- typography https://styled-system.com/table#typography
- color https://styled-system.com/table#color
- space https://styled-system.com/table#space
