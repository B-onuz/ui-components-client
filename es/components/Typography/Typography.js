"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Typography/Typography.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  white-space: pre-line;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledTypography = _styledComponents["default"].p(_templateObject(), (0, _styledSystem.compose)(_styledSystem.typography, _styledSystem.color, _styledSystem.space));

var Typography = function Typography(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/_react["default"].createElement(StyledTypography, _extends({}, rest, {
    children: children,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 11,
      columnNumber: 10
    }
  }));
};

var _default = Typography;
exports["default"] = _default;