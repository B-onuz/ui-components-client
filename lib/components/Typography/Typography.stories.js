'use strict'

exports.__esModule = true
exports.TypographyExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Typography = _interopRequireDefault(require('./Typography'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Typography/Typography.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Typography',
  component: _Typography['default'],
}
exports['default'] = _default

var TypographyExample = function TypographyExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _react['default'].Fragment,
    null,
    /*#__PURE__*/ _react['default'].createElement(
      _Typography['default'],
      {
        color: ['primary', 'olive', 'blue'],
        fontSize: [1, 2, 3, 4],
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 11,
          columnNumber: 5,
        },
      },
      'Minim incididunt incididunt sint enim id nulla aliqua adipisicing tempor. Proident minim mollit Lorem ad qui occaecat magna. Ex ex Lorem duis Lorem in consectetur consectetur incididunt voluptate. Occaecat cillum nisi culpa deserunt. Pariatur dolor consectetur laborum id ipsum elit dolore aute sit reprehenderit culpa occaecat. Adipisicing eu elit in cillum labore occaecat ullamco est ea enim laboris irure dolore. Eu Lorem reprehenderit et ipsum exercitation. Enim quis excepteur anim ea et tempor nisi officia. Eiusmod proident ad amet qui. Dolor nisi do nisi fugiat sint eu veniam occaecat officia. Sint labore officia irure pariatur dolore quis reprehenderit enim sunt aliquip ea ea ad.'
    )
  )
}

exports.TypographyExample = TypographyExample
