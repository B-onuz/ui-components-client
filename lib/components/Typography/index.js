"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _Typography = _interopRequireDefault(require("./Typography"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Typography["default"];
exports["default"] = _default;
module.exports = exports.default;