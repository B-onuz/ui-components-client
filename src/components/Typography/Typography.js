import React from 'react'
import styled from 'styled-components'
import { typography, color, space, compose } from 'styled-system'

const StyledTypography = styled.p`
  ${compose(typography, color, space)}
  white-space: pre-line;
`

const Typography = ({ children, ...rest }) => {
  return <StyledTypography {...rest} children={children} />
}

export default Typography
