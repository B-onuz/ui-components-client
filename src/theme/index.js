const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
