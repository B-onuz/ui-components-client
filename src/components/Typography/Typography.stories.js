import React from 'react'
import Typography from './Typography'

export default {
  title: 'Typography',
  component: Typography,
}

export const TypographyExample = () => (
  <>
    <Typography color={['primary', 'olive', 'blue']} fontSize={[1, 2, 3, 4]}>
      Minim incididunt incididunt sint enim id nulla aliqua adipisicing tempor. Proident minim mollit Lorem ad qui occaecat magna. Ex ex Lorem duis
      Lorem in consectetur consectetur incididunt voluptate. Occaecat cillum nisi culpa deserunt. Pariatur dolor consectetur laborum id ipsum elit
      dolore aute sit reprehenderit culpa occaecat. Adipisicing eu elit in cillum labore occaecat ullamco est ea enim laboris irure dolore. Eu Lorem
      reprehenderit et ipsum exercitation. Enim quis excepteur anim ea et tempor nisi officia. Eiusmod proident ad amet qui. Dolor nisi do nisi fugiat
      sint eu veniam occaecat officia. Sint labore officia irure pariatur dolore quis reprehenderit enim sunt aliquip ea ea ad.
    </Typography>
  </>
)
