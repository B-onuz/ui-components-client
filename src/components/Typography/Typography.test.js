import React from 'react'
import Typography from './Typography.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render Typography', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Typography>Aliquip in nisi ut ipsum.</Typography>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Should render Typography when color prop', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Typography color={'black'} bg={'primary'}>
        Aliquip in nisi ut ipsum.
      </Typography>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Should render Typography when space prop', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Typography m={10}>Aliquip in nisi ut ipsum.</Typography>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
