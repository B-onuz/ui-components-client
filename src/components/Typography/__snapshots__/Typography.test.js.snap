// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Typography 1`] = `
<p
  className=""
>
  Aliquip in nisi ut ipsum.
</p>
`;

exports[`Should render Typography when color prop 1`] = `
.c0 {
  color: #1b1c1d;
  background-color: #ef4d5a;
}

<p
  className="c0"
  color="black"
>
  Aliquip in nisi ut ipsum.
</p>
`;

exports[`Should render Typography when space prop 1`] = `
.c0 {
  margin: 10px;
}

<p
  className="c0"
>
  Aliquip in nisi ut ipsum.
</p>
`;
